File: src/main/java/bootit/incident/IncidentController.java
package bootit.incident;

import bootit.messenger.Messenger;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class IncidentController {

    @Autowired
    private IncidentRepository repository;

    @Autowired
    private Messenger messenger;

    @RequestMapping(value = "/incidents", method = RequestMethod.POST)
    public ResponseEntity<Incident> addIncident(@RequestParam(value="what", defaultValue="") String what) {
        messenger.send("Added new Incident");
        Incident incident = repository.save(new Incident(what));
        return new ResponseEntity<Incident>(incident, HttpStatus.CREATED);
    }

    @RequestMapping(value = "/incidents", method = RequestMethod.GET)
    public Iterable<Incident> getIncidents() {
        return repository.findAll();
    }
}

File: src/main/java/bootit/incident/IncidentController.kt
package bootit.incident

import bootit.messenger.Messenger
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestMethod
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController

@RestController class IncidentController {

    @Autowired
    lateinit var repository: IncidentRepository

    @Autowired
    lateinit var messenger: Messenger

    @RequestMapping(value = "/incidents", method = [RequestMethod.POST])
    fun addIncident(@RequestParam(value = "what", defaultValue = "") what: String) : ResponseEntity<Incident> {
        val incident = repository.save(Incident(what));
        messenger.send("Added new Incident");
        return ResponseEntity(incident, HttpStatus.CREATED);
    }

    @RequestMapping(value = "/incidents", method = [RequestMethod.GET])
    fun getIncidents(@RequestParam(value = "like", defaultValue = "") like: String) = when (like) {
            "" -> repository.findAll()
            else -> repository.findAll().filter { it.what.contains(like) }
    }
}